File: 프로그래머스/1/42889. 실패율/실패율.py
def solution(N, stages):
    answer = []
    
    length = len(stages)
    
    for i in range(1, N+1):
        count = stages.count(i)
        
        # 실패율 계산
        if length == 0:
            fail = 0
        else:
            fail = count / length
        
        # 리스트에 (스테이지번호, 실패율) 삽입
        answer.append((i, fail))
        length -= count
    
    # 실패율 기준으로 정렬
    answer.sort(key=lambda x: -x[1])
    
    # 정렬된 스테이지 번호출력
    answer = [i[0] for i in answer]
    
    return answer
    
        

File: 프로그래머스/1/42889. 실패율/실패율.cpp
#include <bits/stdc++.h>

using namespace std;

double a[504];
vector<pair<int,double>> temp;

bool solve(pair<int,double> a, pair<int,double> b) {
    if (a.second == b.second) return a.first < b.first;
    return a.second > b.second;
}

vector<int> solution(int N, vector<int> stages) {
    vector<int> answer;

    double people = stages.size();

    for (int stage : stages) a[stage]++;

    for (int i=1;i<=N;i++) {
        double rate;
        if (people == 0) rate = 0;
        else {
            rate = a[i] / people;
            people -= a[i];
        }
        

        temp.push_back({i, rate});
    }

    sort(temp.begin(), temp.end(), solve);

    for (pair<int,double> result : temp) answer.push_back(result.first);


    return answer;
}